### PR TITLE
[SvelteKit migration] to latest Sveltekit version

### PR DESCRIPTION
**Pull request by mishig25**
### TLDR: Migrate to latest SvelteKit

The previous SvelteKit that was used in doc-builder was an pre-1.0 release one. There has been a lot of changes on SvelteKit, including new routing mechanism. Overall, it is a good time to migrate to the latest sveltekit and keep `doc-builder` up-to-date

Not sure if there will be any performance gains in building doc html files. If so, that would be nice.

todos:
- [x] remove unnecessary dev files
- [x] python copying mdx files should put them in new sveltekit 1.0 [slug]/page.svelte format bfcaa4573ee68b9f7b3f0b5967f7c61ca628089f
- [x] test it intensively locally (tested it on datasets docs)
- [x] check objects.inv file location being correct
- [x] test it on transformers docs (make sure paths in translated pages are working as expected) https://github.com/huggingface/transformers/pull/26027 [link to workflow](https://github.com/huggingface/transformers/actions/runs/6108206161/job/16576771792)
    - [x] pushed doc file [here](https://huggingface.co/datasets/hf-doc-build/doc-build-dev/blob/main/transformers/pr_26027.zip)
    - [x] newly built doc files causing 500 internal server error (fix this error) (fixed https://github.com/huggingface/doc-builder/pull/396/commits/b0bbce47ba538011f2c194af4d25432c5a20600f)
- [x] title not changing (page should reload on click) (handled in hub pr)
- [x] left side menu not changing (handled in hub pr)
- [x] fix failing test (fixed transformersd dep https://github.com/huggingface/doc-builder/pull/396/commits/bf1b14643479b1356030cc01d10de695f966bf71)
- [x] revert dev doc build action changes

### CODE REVIEW

**Comment by mishig25** [comment]
https://huggingface.co/datasets/hf-doc-build/doc-build-dev/blob/main/transformers/pr_26027.zip
